src/pauli.jl

```diff
@@ -0,0 +1,229 @@
+    pbv = pauli_basis_vectors(num_qubits)
+    sop_dim = 2 ^ (2 * num_qubits)
+    data = Array{Complex{Float64}}(undef, (sop_dim, sop_dim))
+    data .= pbv * ptm.data * pbv' / √sop_dim
+    return DenseSuperOperator(ptm.basis_l, ptm.basis_r, data)
+end
+
+"""
+    PauliTransferMatrix(unitary::DenseOperator)
+
+Convert an operator, presumably a unitary operator, to its representation as a
+Pauli transfer matrix.
+"""
+PauliTransferMatrix(unitary::DenseOperator{B, B, Array{Complex{Float64},2}}) where B <: PauliBasis = PauliTransferMatrix(SuperOperator(unitary))
+
+"""
+    ChiMatrix(unitary::DenseOperator)
+
+Convert an operator, presumably a unitary operator, to its representation as a χ matrix.
+"""
+function ChiMatrix(unitary::DenseOperator{B, B, Array{Complex{Float64},2}}) where B <: PauliBasis
+    num_qubits = length(unitary.basis_l.bases)
+    pbv = pauli_basis_vectors(num_qubits)
+    aj = pbv' * reshape(unitary.data, 2 ^ (2 * num_qubits))
+    return DenseChiMatrix((unitary.basis_l, unitary.basis_l), (unitary.basis_r, unitary.basis_r), aj * aj' / (2 ^ num_qubits))
+end
+
+"""
+    ChiMatrix(sop::DenseSuperOperator)
+
+Convert a superoperator to its representation as a Chi matrix.
+"""
+function ChiMatrix(sop::DenseSuperOperator{B, B, Array{Complex{Float64}, 2}}) where B <: Tuple{PauliBasis, PauliBasis}
+    num_qubits = length(sop.basis_l)
+    sop_dim = 2 ^ (2 * num_qubits)
+    po = pauli_operators(num_qubits)
+    data = Array{Complex{Float64}, 2}(undef, (sop_dim, sop_dim))
+    for (idx, jdx) in Iterators.product(1:sop_dim, 1:sop_dim)
+        data[idx, jdx] = tr((spre(po[idx]) * spost(po[jdx])).data' * sop.data) / √sop_dim
+    end
+    return DenseChiMatrix(sop.basis_l, sop.basis_r, data)
+end
+
+"""
+    PauliTransferMatrix(chi_matrix::DenseChiMatrix)
+
+Convert a χ matrix to its representation as a Pauli transfer matrix.
+"""
+function PauliTransferMatrix(chi_matrix::DenseChiMatrix{B, B, Array{Complex{Float64}, 2}}) where B <: Tuple{PauliBasis, PauliBasis}
+    num_qubits = length(chi_matrix.basis_l)
+    sop_dim = 2 ^ (2 * num_qubits)
+    po = pauli_operators(num_qubits)
+    data = Array{Float64, 2}(undef, (sop_dim, sop_dim))
+    for (idx, jdx) in Iterators.product(1:sop_dim, 1:sop_dim)
+        data[idx, jdx] = tr(mapreduce(x -> po[idx] * po[x[1]] * po[jdx] * po[x[2]] * chi_matrix.data[x[1], x[2]],
+                                      +,
+                                      Iterators.product(1:16, 1:16)).data) / sop_dim |> real
+    end
+    return DensePauliTransferMatrix(chi_matrix.basis_l, chi_matrix.basis_r, data)
+end
+
+"""
+    SuperOperator(chi_matrix::DenseChiMatrix)
+
+Convert a χ matrix to its representation as a superoperator.
+"""
+function SuperOperator(chi_matrix::DenseChiMatrix{B, B, Array{Complex{Float64}, 2}}) where B <: Tuple{PauliBasis, PauliBasis}
+    return SuperOperator(PauliTransferMatrix(chi_matrix))
+end
+
+"""
+    ChiMatrix(ptm::PauliTransferMatrix)
```

Comment:
Should this also say Dense?

Reply:
yes. i also wonder if the constructors themselves shouldn't be called Dense...